src/marten/db/query/prefetcher.cr

```diff
@@ -166,20 +171,32 @@ module Marten
           records_to_decorate
         end
 
+        private def prefetch_query_set(relation_name, default_model)
+          if custom_query = @custom_query_sets[relation_name]?
+            raise Errors::MismatchedQuerySetType.new(
+              "Can't prefetch :#{relation_name} using #{custom_query.model} query set."
+            ) if custom_query.model != default_model
+            custom_query
+          else
+            default_model.unscoped
```

Comment:
Just some thought:

Now that Marten provides the functionality to use any custom query, what do you think about reverting this to use the default scope?

For example if a `Board` has many users, which have a column `deleted_at`, it would be helpful to only load those that are not deleted.

I know this is a breaking change, but it makes more sense to me

Reply:
For now I would say let's keep using `unscoped`. Your example makes sense when considering the case of soft-deletion, but default scopes can be overridden for things that are completely different. I don't know if/when we will add soft-deletion support to the framework, so in the meantime I don't think there is need for changing this default behavior.